src/Forms/ConfirmedPasswordField.php

```diff
@@ -37,6 +47,11 @@ class ConfirmedPasswordField extends FormField
      */
     public int $maxLength = 0;
 
+    /**
+     * Whether the field is on a member form
+     */
+    private bool $isOnMemberForm = false;
```

Comment:
The edit form in the CMS (both in `admin/security` and in `admin/myprofile`) already has a reference to the member record which we can fetch from the field with `$this->getForm()->getRecord()`.

If we set the member record into the change password form we could remove this `isOnMemberForm` property and its setter method, and instead check `is_a($this->getForm()?->getRecord(), Member::class)`.

I'm not terribly fussed about this so feel free to keep as is if you prefer this setter pattern.

Reply:
It will work if you do what I suggested
> If we set the member record into the change password form

As I said, the form _in the CMS_ has the record set. The change password form _does not_ so to use this implementation you'd have to set the record in the form.

Reply:
Opps sorry missed that

Still won't work, in the "lost password flow" when you click an email `Security::getCurrentUser()` will return null i.e. we could never call `$form->loadDataFrom($member)`

Reply:
We know who the hash belongs to though.
It will work if you want to do it, and I can help if you want to explore it. If you'd rather not, that's fine too.

Reply:
I'd rather not